### DISCUSSION TITLE
[@types/node] module loader hooks type

### DISCUSSION THREAD

**Discussion by himself65** [DefinitelyTyped/DefinitelyTyped#65431] [Issues with a @types package]
https://nodejs.org/dist/latest-v18.x/docs/api/esm.html#loaders

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
@Himself65 Do you plan to use it for writing own custom loaders? Can you provide feedback before merging if I will provide types for you?

**Reply by himself65** [reply]
Yes

**Reply by himself65** [reply]
Here is a real code which are using hooks
https://github.com/dai-shi/wakuwork/blob/main/src/node-loader.ts

**Reply by Semigradsky** [reply]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65490
